### DISCUSSION TITLE
Resetting Stop Flag

### DISCUSSION THREAD

**Discussion by jacob-kruse** [kroshu/kuka_drivers#200] [Q&A] [answered]
Hello,

I am utilizing the drivers in this package to control a KUKA LBR iisy r760 with MoveIt2's Python API MoveItPy. Sometimes if the velocity/acceleration scaling factor is set too high in my configuration for MoveIt, the velocity/acceleration limits will be exceeded, leading to an Error and the protective stop engaging. I have done some digging into the KukaEACHardware and kuka-external-control-sdk and have found no way to turn off the stop flag that is raised without killing and restarting my terminal session. I was wondering if it is possible right now to reset this with a service or call to the hardware interface. If not, is there an easy way to implement this functionality.

Thanks in advance,
Jacob Kruse

**Comment by Svastits** [comment]
Hi, 

If the limits are exceeded and the robot stops, the `robot_manager` node should get deactivated, after which activating it again (`ros2 lifecycle set robot_manager activate`) should start control again without having to kill the process

**Reply by jacob-kruse** [reply]
Thanks for the quick response! But, this is not the case for me. When the robot stops, the robot manager node is still active. I have tried what you suggested, but since the robot manager is still active, nothing changes. The exact error I get is: "External control stopped by an error". It seems this error message is published by event_observer.hpp which calls set_stop_flag() in hardware_interface.hpp after observing an error. Any other suggestions?

**Reply by Svastits** [reply, marked as answer]
Oh, I understand, in this case this is a bug, as every error message should deactivate the robot manager. Please create a bugticket with steps to reproduce and we will investigate :)

**Reply by jacob-kruse** [reply]
Okay, sounds good. I have submitted the issue.

See Issue #202
